verify/src/main/java/com/twilio/verify/networking/NetworkProvider.kt

```diff
@@ -42,8 +46,30 @@ class NetworkException constructor(
   )
 }
 
+internal const val CODE_KEY = "code"
+internal const val MESSAGE_KEY = "message"
+internal const val MORE_INFO_KEY = "more_info"
+
 class FailureResponse(
-  val responseCode: Int,
+  val statusCode: Int,
   val errorBody: String?,
   val headers: Map<String, List<String>>?
+) {
+  val apiError: APIError? by lazy {
+    return@lazy errorBody?.let {
+      try {
+        val json = JSONObject(it)
+        return@let APIError(json.getString(CODE_KEY), json.getString(MESSAGE_KEY), json.optString(MORE_INFO_KEY))
+      } catch (e: Exception) {
+        Logger.log(Level.Networking, "Unable to convert error body to Verify API error, details: $e")
+        return@let null
+      }
+    }
+  }
+}
+
+class APIError(
+  val code: String,
+  val message: String,
+  val moreInfo: String
```

Comment:
I set this as optional in the iOS implementation, should this class has the optional too? wdyt?

Reply:
Fixed, thanks